OctaneVSPlugin/Common/EntityComparerByLastModified.cs

```diff
@@ -18,45 +18,38 @@
 using System;
 using System.Collections.Generic;
 
-namespace MicroFocus.Adm.Octane.VisualStudio
+namespace MicroFocus.Adm.Octane.VisualStudio.Common
 {
     /// <summary>
-    /// Comparer for BaseEntity by LastModified field.
+    /// Comparer for BaseEntity by LastModified field (only in descending mode)
     /// </summary>
     internal class EntityComparerByLastModified : IComparer<BaseEntity>
     {
-        // For now we always order in descending mode.
-        private const bool isDescending = true;
-
-        public int Compare(BaseEntity x, BaseEntity y)
+        /// <inheritdoc/>
+        public int Compare(BaseEntity first, BaseEntity second)
         {
-            DateTime? xTime = x.GetDateTimeValue(CommonFields.LAST_MODIFIED);
-            DateTime? yTime = y.GetDateTimeValue(CommonFields.LAST_MODIFIED);
-            int result;
+            if (first == null)
+                throw new ArgumentNullException(nameof(first));
+            if (second == null)
+                throw new ArgumentNullException(nameof(second));
 
-            if (!xTime.HasValue && !yTime.HasValue)
-            {
-                result = 0;
-            }
-            else if (!xTime.HasValue)
-            {
-                result = 1;
-            }
-            else if (!yTime.HasValue)
+            DateTime? firstDateTime = first.GetDateTimeValue(CommonFields.LastModified);
+            DateTime? secondDateTime = second.GetDateTimeValue(CommonFields.LastModified);
+
+            if (!firstDateTime.HasValue && !secondDateTime.HasValue)
             {
-                result = 0;
+                return 0;
             }
-            else
+            if (!firstDateTime.HasValue)
             {
-                result = xTime.Value.CompareTo(yTime.Value);
+                return -1;
             }
-
-            if (isDescending)
+            if (!secondDateTime.HasValue)
             {
-                result = -1 * result;
+                return 1;
             }
 
-            return result;
+            return -1 * firstDateTime.Value.CompareTo(secondDateTime.Value);
```

Comment:
Why is it needed to reverse the sign of the result of CompareTo?

Reply:
@kdrtibor Because we want to sort by descending date

Reply:
@paul-ciobanu  but why not switch only the order of comparison?

Reply:
@kdrtibor good point. fixed